packages/@react-aria/menu/src/useSubMenuTrigger.ts

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright 2023 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+
+import {AriaMenuItemProps} from './useMenuItem';
+import {AriaMenuOptions} from './useMenu';
+import type {AriaPopoverProps} from '@react-aria/overlays';
+import {FocusableElement, FocusStrategy, PressEvent, Node as RSNode} from '@react-types/shared';
+import {RefObject, useCallback, useRef} from 'react';
+import type {SubMenuTriggerState} from '@react-stately/menu';
+import {useEffectEvent, useId, useLayoutEffect} from '@react-aria/utils';
+import {useLocale} from '@react-aria/i18n';
+
+export interface AriaSubMenuTriggerProps {
+  /** An object representing the submenu trigger menu item. Contains all the relevant information that makes up the menu item. */
+  node: RSNode<unknown>,
+  /** Whether the submenu trigger is disabled. */
+  isDisabled?: boolean,
+  // TODO: naming. Also talk about if this should be customizable in this hook or if it belongs somewhere else
+  /** Type of the submenu being rendered. */
+  subMenuType?: 'dialog' | 'menu',
```

Comment:
Naming of this prop? I assume unavailable menu items is something we'll want to support in aria as well